test/integration/action_execution.go

```diff
@@ -490,6 +515,25 @@ func waitForDeployment(client kubeclientset.Interface, depName, namespace string
 	return newDep, nil
 }
 
+func waitForDeploymentToUpdateResource(client kubeclientset.Interface, depName, namespace string, oldResourceVersion string) (*appsv1.Deployment, error) {
```

Comment:
This can make the test flaky.
Controllers can sometimes update resources, resulting in resource version updates.
I would rather have a the wait on what we are expecting as the result of testing here. 
`waitForResourcesUpdated()` where the wait will be on the `dep.Spec.Template.Spec.Containers[0].Resources` to be matched to what we expect after update. This also means that `validateResourceResize` will not be needed.

Reply:
@irfanurrehman  I removed the function `validateResourceResize` and put the evaluation into the wait function as suggested, please have a  look at my new commit. Thanks for your review.